src/front/Pages/Index.razor

```diff
@@ -1,7 +1,25 @@
 ﻿@page "/"
+@inject HttpClient Http
 
 <PageTitle>Index</PageTitle>
 
-<h1>Hello, world!</h1>
+@if (_root == null)
+{
+    <div class="loading-container full-page">
+        <div class="spinner-grow text-primary" role="status">
+            <span class="visually-hidden">Loading...</span>
+        </div>
+    </div>
+}
+else
+{
+    <YamlNode Node="@_root" />
+}
 
-Welcome to your new app.
+@code {
+    private YamlNodeModel? _root = null;
+    protected override async Task OnInitializedAsync()
+    {
+        _root = await Http!.GetFromJsonAsync<YamlNodeModel>("sample-data/yaml-tree.json");
```

Comment:
Yup, that makes sense. I wanted to focus on the UI design first, but as you mention, mock data should use similar service than real data, this will definitely change once you provide the service.